lib/Cake/View/Helper/FormHelper.php

```diff
@@ -2866,7 +2866,7 @@ protected function _selectOptions($elements = array(), $parents = array(), $show
 					}
 				}
 
-				if ($showParents || (!in_array($title, $parents))) {
+				if ($showParents || (!in_array($title, $parents, true))) {
```

Comment:
That will probably not be BC with some people's apps, as this will now also not work for int values coming in and are expected to be used as values here.